prototype2/gdgem/nmx/Readout.h

```diff
@@ -1,40 +1,98 @@
 /* Copyright (C) 2017-2018 European Spallation Source, ERIC. See LICENSE file */
 //===----------------------------------------------------------------------===//
 ///
+///                              WARNING
+///
+///                              ACHTUNG
+///
+///                              AVISO
+///
+///                              ADVARSEL
+///
+///                              DĖMESIO
+///
+///                              UWAGA
+///
+///
+///
+///          MODIFY THIS FILE ONLY AFTER UNDERSTANDING THE FOLLOWING
+///
+///
+///   Any changes to non-static variable definitions will likely break h5 file
+/// writing and compatibility. If you rename, reorder or change the type of any
+/// of the member variables, you MUST:
+///    A) Increment FormatVersion by 1
+///    B) Ensure the hdf5 TypeTrait maps the struct correctly
+///
+/// If you cannot ensure the above, consult someone who can.
+///
 /// \file
 ///
+/// \brief Readout struct for NMX event formation
+///
 //===----------------------------------------------------------------------===//
 
 #pragma once
 
-#include <cinttypes>
-
-// TODO: initialize values to 0?
-struct Readout
-{
-  uint8_t fec;
-  uint8_t chip_id;
-  uint32_t bonus_timestamp;
-  uint32_t srs_timestamp;
-  uint16_t channel;
-  uint16_t bcid;
-  uint16_t tdc;
-  uint16_t adc;
-  float ChipTimeNs;
-  bool over_threshold;
-
-  bool operator==(const Readout& other) const
-  {
+#include <common/DumpFile.h>
+
+struct __attribute__ ((packed)) Readout {
```

Comment:
What do the fields mean, are they documented somewhere?

Reply:
This is all SRS/VMM readout system specific stuff. How it all works is one of the great mysteries of the universe. If we can get someone at CERN to properly document it, I buy everyone a beer.